### PR TITLE
DBAAS-131 Implement changes to align with DBaaSProvider API changes

### PR DESCRIPTION
**Pull request by jianrongzhang89**
Implement the changes for DBaaSProvider. Clean up previous configmap based dbaas registration logic.

Tested in local CRC env successfully.

### CODE REVIEW

File: config/dbaasprovider/dbaas_provider.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+metadata:
+  name: mongodb-atlas-registration
+  labels:
+    related-to: dbaas-operator
+    type: dbaas-provider-registration
+spec:
+  provider:
+    name: Red Hat DBaaS / MongoDB Atlas
+    displayName: MongoDB Atlas Cloud Database Service
+    displayDescription: MongoDB Atlas Fully Managed Database Service.
```
Comment (redhatHameed) [inline config/dbaasprovider/dbaas_provider.yaml:12]: not an issue, just to make sure to discuss this with Mondo DB if the display description is the right one or you can have a look on their website if required any change in that. 
https://www.mongodb.com/cloud/atlas/lp/try2?utm_source=google&utm_campaign=gs_americas_united_states_search_core_brand_atlas_desktop&utm_term=mongodb%20atlas%20login&utm_medium=cpc_paid_search&utm_ad=e&utm_ad_campaign_id=12212624338&gclid=EAIaIQobChMI7efI4PyK8gIVI4VaBR3tOAHmEAAYASAAEgIMQvD_BwE

**Review by redhatHameed — APPROVED** [review]

File: pkg/api/dbaas/info.go [hunk 2]
```diff
@@ -26,13 +26,13 @@ import (
...
 
 	// SchemeBuilder is used to add go types to the GroupVersionKind scheme
-	DBaaSSchemeBuilder = &scheme.Builder{GroupVersion: DBaaSGroupVersion}
+	SchemeBuilder = &scheme.Builder{GroupVersion: GroupVersion}
 
 	// AddToScheme adds the types in this group-version to the given scheme.
-	AddToScheme = DBaaSSchemeBuilder.AddToScheme
+	AddToScheme = SchemeBuilder.AddToScheme
 
 	_ = &MongoDBAtlasInventory{}
```
Comment (jeremyary) [inline pkg/api/dbaas/info.go:37]: It won't let me select collapsed lines here to comment on, but what's the purpose of these?
```
  _ = &MongoDBAtlasInventory{}

	_ = &MongoDBAtlasConnection{}
```

File: config/dbaasprovider/dbaas_provider.yaml (hunk 1 above)
Comment (jianrongzhang89) [reply config/dbaasprovider/dbaas_provider.yaml:12]: This one is better :). Change to "Cloud-hosted MongoDB service on AWS, Azure and Google Cloud"

File: pkg/api/dbaas/info.go (hunk 2 above)
Comment (jianrongzhang89) [reply pkg/api/dbaas/info.go]: Good catch - they were from the original implementation and are no longer needed. I cleaned them up.

**Pull request merged**